src/nvim/main.c

```diff
@@ -2057,93 +2057,66 @@ static void mainerr(int n, const char *str)
-  main_msg(_("-r\t\t\tList swap files and exit"));
-  main_msg(_("-r (with file name)\tRecover crashed session"));
-  main_msg(_("-L\t\t\tSame as -r"));
-  main_msg(_("-A\t\t\tstart in Arabic mode"));
-  main_msg(_("-H\t\t\tStart in Hebrew mode"));
-  main_msg(_("-F\t\t\tStart in Farsi mode"));
-  main_msg(_("-T <terminal>\tSet terminal type to <terminal>"));
-  main_msg(_("-u <vimrc>\t\tUse <vimrc> instead of any .vimrc"));
-  main_msg(_("--noplugin\t\tDon't load plugin scripts"));
-  main_msg(_("-p[N]\t\tOpen N tab pages (default: one for each file)"));
-  main_msg(_("-o[N]\t\tOpen N windows (default: one for each file)"));
-  main_msg(_("-O[N]\t\tLike -o but split vertically"));
-  main_msg(_("+\t\t\tStart at end of file"));
-  main_msg(_("+<lnum>\t\tStart at line <lnum>"));
-  main_msg(_("--cmd <command>\tExecute <command> before loading any vimrc file"));
-  main_msg(_("-c <command>\t\tExecute <command> after loading the first file"));
-  main_msg(_(
-        "-S <session>\t\tSource file <session> after loading the first file"));
-  main_msg(_("-s <scriptin>\tRead Normal mode commands from file <scriptin>"));
-  main_msg(_("-w <scriptout>\tAppend all typed commands to file <scriptout>"));
-  main_msg(_("-W <scriptout>\tWrite all typed commands to file <scriptout>"));
-  main_msg(_("--startuptime <file>\tWrite startup timing messages to <file>"));
-  main_msg(_("-i <viminfo>\t\tUse <viminfo> instead of .viminfo"));
-  main_msg(_("--api-info\t\tDump API metadata serialized to msgpack and exit"));
-  main_msg(_("--embed\t\tUse stdin/stdout as a msgpack-rpc channel. "
-             "This can be used for embedding Neovim into other programs"));
-  main_msg(_("-h  or  --help\tPrint Help (this message) and exit"));
-  main_msg(_("--version\t\tPrint version information and exit"));
+  mch_msg(_("  -v                    Vi mode (like \"vi\")\n"));
+  mch_msg(_("  -e                    Ex mode (like \"ex\")\n"));
+  mch_msg(_("  -E                    Improved Ex mode\n"));
+  mch_msg(_("  -s                    Silent (batch) mode (only for \"ex\")\n"));
+  mch_msg(_("  -d                    Diff mode (like \"vimdiff\")\n"));
+  mch_msg(_("  -R                    Readonly mode (like \"view\")\n"));
+  mch_msg(_("  -Z                    Restricted mode (like \"rvim\")\n"));
+  mch_msg(_("  -m                    Modifications (writing files) not allowed\n"));
+  mch_msg(_("  -M                    Modifications in text not allowed\n"));
+  mch_msg(_("  -b                    Binary mode\n"));
+  mch_msg(_("  -l                    Lisp mode\n"));
+  mch_msg(_("  -V[N][file]           Be verbose [level N][log messages to file]\n"));
+  mch_msg(_("  -D                    Debugging mode\n"));
+  mch_msg(_("  -n                    No swap file, use memory only\n"));
+  mch_msg(_("  -r                    List swap files and exit\n"));
+  mch_msg(_("  -r <file>             Recover crashed session\n"));
+  mch_msg(_("  -L                    Same as -r\n"));
```

Comment:
`-L` appears to be a vi equivalent. I don't know why `-r` exists also, but since `-L` does the same thing, there is zero reason to mention it in the already-very-busy `Usage:` message. Of course, we will still continue to support `-L` and document it in `:help`. 

But for `-h` message, eliminating noise is a high priority.